frontend/ui/widget/keyboardlayoutdialog.lua

```diff
@@ -58,15 +58,26 @@ function KeyboardLayoutDialog:init()
 
     local buttons = {}
     local radio_buttons = {}
+
+    local keyboard_layouts = G_reader_settings:readSetting("keyboard_layouts") or {}
+    G_reader_settings:makeTrue("keyboard_force_current_layout")
```

Comment:
I understand this is some kind of temporary flag (I'll never see it in my settings.reader.lua when exiting KOreader), is it ? Feels odd to store it in G_reader_settings. No other common accessible object from here and the other module that you could just store it in ?

Anyway, this needs some comment here as for why we do that.
edit: oh, you have that comment below in the other file :)

Reply:
Can you give an idea where to save this flag instead of the settings?

Reply:
Frankly, this reinit https://github.com/koreader/koreader/commit/397211644de48eb87742152a74c8f112f0b9d1cf is a little bit a PITA, destroying checkbuttons in inputdialog, requiring this trick with a flag.

Reply:
I would have said the thing we can reach from nearly everywhere :) `UIManager._keyboard_force_current_layout`.

But may be it can be kept as a local in virtualkeyboard.lua ?
Not tested:
Somewhere near the top:
```lua
local keyboard_state = {
    force_current_layout = false,
}
```
(a table, so we can pass it and update its content - a simple variable would not allow that).

And pass it to your two `KeyboardLayoutDialog:new`:
```lua
            self.keyboard_layout_dialog = KeyboardLayoutDialog:new{
                parent = self,
                keyboard_state = keyboard_state,
            }
```

Reply:
> a local in virtualkeyboard.lua

Will it survive after?
https://github.com/koreader/koreader/blob/39b810a3bdd3dacbfd1811cb88e111c37204b5a8/frontend/ui/widget/inputdialog.lua#L533-L534

Reply:
The logic of this flag: to dictinguish kbd inits
(1) new kbd (honor settings only)
(2) change layout (force selected layout)
But (2) if "kbd height changed" makes (1) thus this flag.

Reply:
> Will it survive after?

It should: it's a local to the module, a module is loaded only once, so it's created/inited once.

(You could even put it as a "class member" - it being a table and it never being reset in :init(), this same table would be accessible to all instances - the risky stuff I talked about in the FileSearcher issue could have some useful use here - but a module local makes it less risky and needs less documentation :)

Reply:
Thanks, I'll try.

Reply:
Yes, it works, thanks a lot.